src/Compilers/Core/MSBuildTask/MapSourceRoots.cs

```diff
@@ -24,6 +24,8 @@ public sealed class MapSourceRoots : Task
     {
         public MapSourceRoots()
         {
+            SourceRoots = null!; // PROTOTYPE(NullableDogfood): Required by MSBuild
```

Comment:
> Required by MSBuild

📝 Saw this a few times; wasn't quite sure what it means

Reply:
The `SourceRoots` property has an attribute `[Required]` (from MSBuild). It means that MSBuild will check this for us and we can assume that it is already initialized.
The constructor however doesn't know that.